### ISSUE TITLE
private(set) causes a private_outlet Private Outlets Violation warning

### ISSUE THREAD

**Issue by magesteve** [realm/SwiftLint#1999]
### Bug Report

The private_outlet rule returns a warning when using the "private(set)" scope.  This is a valid form of private scope, and as such, should not return a warning.

### Environment

* SwiftLint version: 0.24.0
* Installation method used: brew
* XCode version: 9.2 (but have had this issue with earlier versions of XCode).
* Configuration file: Any that file that sets the "private_outlet" rule.

**Comment by jpsim** [comment]
Could you please fill out the [issue template](https://github.com/realm/SwiftLint/blob/master/.github/ISSUE_TEMPLATE.md) so we have enough information to help you? Thanks!

---

### Bug Report

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcode-select -p`)?
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
let foo = try! bar()
```

**Comment by magesteve** [comment]
@jpsim This is not a bug in the sense of a crash in the program, which would require a stack trace or the command used.   I did not post the configuration file, since it would occur any time you have the private_outlet set.  The rule requires a IBOutlet property to have it's scope set to "private" (which is a good idea), but does not allow it to be set to "private(set)". Private(set) is valid version of the private scope that has not been used in the past, but is gaining popularity. It allows other Controllers to access and configure the IBOutlet of another view controller, while not allowing it to set it. Usually IBOutlets are only set by being instantiates as part of an Interface file.

**Comment by ornithocoder** [comment]
I like the way the current implementation works where private means private, hiding the property from the external world completely, making it an implementation detail rather than part of the object's interface.

Making a property private allows 1) to protect it from being replaced and 2) from property mutation (by external entities). `private(set)` does protect against the former, but not against the latter.

If the issue described gets implemented, I would suggest to make it configurable, so the current behavior is maintained for those who are used to it or prefer the property to stay out of the object interface. It would be something like this:

```
private_outlet:
    allows_public_getter: true
```

I can try to work on it, if you all agree.

**Comment by magesteve** [comment]
Ornithocoder, I see your point. I usually prefer to make fill calls to a View Controller, and let the controller directly access the IBoutlet view.  However, more and more with massive controllers, I have seen code where outside controllers want to access another viewcontrollers fields.  I like your idea of making it an option, to what level of control you want.

**Comment by ornithocoder** [comment]
Hi @magesteve, I checkout the code to start working on it and noticed the [support's already there](https://github.com/realm/SwiftLint/blob/master/Source/SwiftLintFramework/Rules/RuleConfigurations/PrivateOutletRuleConfiguration.swift#L28). It's called `allow_private_set` and it takes a boolean as parameter.

**Comment by marcelofabri** [comment]
Hi @magesteve, as @ornithocoder said, this is already supported via a configuration.

I'm closing this for know. Feel free to reopen if you can't make the configuration work or if you have more questions.

**Comment by magesteve** [comment]
No, the problem was solved. Thanks for the help!

**Comment by User2004** [comment]
Set Outlet collection of any controller in swift. [Click here](https://www.webappcodes.com/how-to-set-outlet-collections-on-ios-controller-in-swift) for view example code.
